homework_4/numpy_challenge.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+if __name__ == "__main__":
+   array1_tarrantion_movies = np.array(['Reservoir Dogs ', 'Pulp Fiction', 
+'Jackie Brown', 'Kill Bill: Vol. 1', 'Kill Bill: Vol. 2', 'Death Proof', 
+'Inglourious Basterds',  'Django Unchained', 'The Hateful Eight', 'Once 
+Upon a Time in Hollywoo'])
+   array2 = np.eye(5)
+   array3 = np.linspace(0, 25, 300)
+
+
+
+def matrix_multiplication(matrix1, matrix2):
+    try:
+        result = np.multiply(matrix1, matrix2)
+        return result
+    except ValueError:
+        return print('does not satisfy the condition of matrix 
+multiplication')
+
+def multiplication_check(matrix_list):
+  for i in range(len(matrix_list) - 1):
+    if matrix_list[i].ndim != 2 or len(matrix_list) < 2:
+      return False
+      break
+    if matrix_list[i+1].ndim != 2:
+      return False
+      break
+    rows1, columns1 = matrix_list[i].shape
+    rows2, columns2 = matrix_list[i+1].shape
+    if columns1 == rows2:
+      continue
+    else:
+      return False
+      break
+  return True
```

Comment:
Что-то как-то сложно, можно сильно упростить
```suggestion
  for i in range(len(matrix_list) - 1):
    if matrix_list[i].shape[1] != matrix_list[i + 1].shape[0]:
        return False
  return True
```